react/PWAContext.js

```diff
@@ -0,0 +1,52 @@
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+} from 'react'
+
+const PWAContext = React.createContext(null)
+
+const PWAProvider = ({ settings, children }) => {
+  const deferredPrompt = useRef(null)
+  /* beforeinstallprompt event is fired even after the userChoice is to cancel (and there is no need to re-render) */
+  const captured = useRef(false)
+
+  useEffect(() => {
+    const handleBeforeInstall = e => {
+      const { promptOnCustomEvent } = settings
+      if (promptOnCustomEvent && !captured.current) {
+        e.preventDefault()
+        deferredPrompt.current = e
+        captured.current = true
+        return false
+      }
+      return true
+    }
+
+    window.addEventListener('beforeinstallprompt', handleBeforeInstall)
+    return () =>
+      window.removeEventListener('beforeinstallprompt', handleBeforeInstall)
+  }, [captured, settings])
+
+  const showInstallPrompt = useCallback(() => {
+    const prompt = deferredPrompt.current
+    if (prompt) {
+      prompt.prompt()
+      prompt.userChoice.finally(result => {
```

Comment:
Our polyfill doesnt include `finally`:

> https://cdn.polyfill.io/v2/polyfill.min.js?features=Object.values,Object.entries,Object.assign,Array.from,Array.prototype.findIndex,Array.prototype.fill,Array.prototype.find,Array.prototype.includes,String.prototype.startsWith,String.prototype.endsWith,String.prototype.includes,Symbol,fetch,Promise,Map,Set,Function.name,Intl.~locale.pt-BR&flags=gated

So this won't work on IE11:
https://caniuse.com/#feat=promise-finally

Reply:
But I guess this will never run in IE11. No PWA there.

Reply:
Yep. It won't run. Should I change this?